android/anc/src/main/java/org/smartregister/fhircore/anc/data/AncPatientRepository.kt

```diff
@@ -108,4 +118,56 @@ class AncPatientRepository(
       }
     return listCarePlan
   }
+
+  suspend fun enrollIntoAnc(patient: Patient) {
+    val pregnancyCondition = loadConfig(Template.PREGNANCY_CONDITION, Condition::class.java)
+    pregnancyCondition.apply {
+      this.id = getUniqueId()
+      this.subject = patient.asReference()
+      this.onset = DateTimeType.now()
+    }
+    fhirEngine.save(pregnancyCondition)
+
+    val pregnancyEpisodeOfCase =
+      loadConfig(Template.PREGNANCY_EPISODE_OF_CARE, EpisodeOfCare::class.java)
+    pregnancyEpisodeOfCase.apply {
+      this.id = getUniqueId()
+      this.patient = patient.asReference()
+      this.diagnosis[0].condition = pregnancyCondition.asReference()
+      this.period = Period().apply { this@apply.start = Date() }
+      this.status = EpisodeOfCare.EpisodeOfCareStatus.ACTIVE
+    }
+    fhirEngine.save(pregnancyEpisodeOfCase)
+
+    val pregnancyEncounter = loadConfig(Template.PREGNANCY_FIRST_ENCOUNTER, Encounter::class.java)
+    pregnancyEncounter.apply {
+      this.id = getUniqueId()
+      this.status = Encounter.EncounterStatus.INPROGRESS
+      this.subject = patient.asReference()
+      this.episodeOfCare = listOf(pregnancyEpisodeOfCase.asReference())
+      this.period = Period().apply { this@apply.start = Date() }
+      this.diagnosis[0].condition = pregnancyCondition.asReference()
+    }
+    fhirEngine.save(pregnancyEncounter)
+
+    val pregnancyGoal =
+      Goal().apply {
+        this.id = getUniqueId()
+        this.lifecycleStatus = Goal.GoalLifecycleStatus.ACTIVE
+        this.subject = patient.asReference()
+      }
+    fhirEngine.save(pregnancyGoal)
+  }
+
+  private fun <T : Any?> loadConfig(id: String, clazz: Class<T>): T {
```

Comment:
if this method will return a resource then we can bound the generic and use `Resource?` instead of `Any?`